__mocks__/react-native-onyx.js

```diff
@@ -0,0 +1,24 @@
+import Onyx, {withOnyx} from 'react-native-onyx';
+
+let connectCallbackDelay = 0;
+function addDelayToConnectCallback(delay) {
+    connectCallbackDelay = delay;
+}
+
+export default {
+    ...Onyx,
+    connect: mapping => Onyx.connect({
+        ...mapping,
+        callback: (params) => {
+            if (connectCallbackDelay > 0) {
+                setTimeout(() => {
+                    mapping.callback(params);
+                }, connectCallbackDelay);
+            } else {
+                mapping.callback(params);
+            }
+        },
+    }),
+    addDelayToConnectCallback,
+};
+export {withOnyx};
```

Comment:
Hmm I liked the mock we had before. Why does it not work?

Reply:
Yeah, I was trying to make it work with AsyncStorage mock but I found a rabbit hole when it's used with Onyx. The main problem is that `API.js` already calls `Onyx.connect` before the unit tests are set up, which makes it a little difficult to have control of those `Onyx.connect` calls. For example in some cases `matchingKeys` were empty (I think because of Onyx.clear in beforeEach) and in other cases  the`Onyx.connect` callbacks registered at the top of `API.js` were already cached and not being resolved by the `AsyncStorage.getItem` function

https://github.com/Expensify/react-native-onyx/blob/224e2ba03f2365726b8c1d0785e750c54a85332b/lib/Onyx.js#L42-L45

Maybe an easier solution (not for now) could be an `init()` function in `API.js`, which register the `Onyx.connect` callbacks, and we can call it after preparing / setting up the unit tests.

Reply:
Ah too bad. I guess it's OK? I found this a little confusing because we are not adding the throttle to the thing that we suspect is going slow, but to `Onyx.connect()` which can't really be slow.

Reply:
Yeah, I think mocking only `Onyx.connect()` is ok atm to only test the network retry when storage is not ready. Hopefully, we can revise this test later and mock `AsyncStorage. getItem` after agreeing something in the `Onyx.connect` [discussion](https://github.com/Expensify/App/issues/6151).